pkg/daemon/rpm-ostree.go

```diff
@@ -194,7 +195,8 @@ func (r *RpmOstreeClient) GetBootedOSImageURL() (string, string, error) {
 			osImageURL = tokens[1]
 		}
 	}
-	return osImageURL, bootedDeployment.Version, nil
+
+	return osImageURL, bootedDeployment.Version, bootedDeployment.Checksum, nil
```

Comment:
When client-side changes are involved (layering/overrides), the checksum will differ.  I think what we want here is to use 

```
diff --git a/pkg/daemon/rpm-ostree.go b/pkg/daemon/rpm-ostree.go
index 76432b60b..3cb797394 100644
--- a/pkg/daemon/rpm-ostree.go
+++ b/pkg/daemon/rpm-ostree.go
@@ -38,6 +38,7 @@ type RpmOstreeDeployment struct {
        OSName                  string   `json:"osname"`
        Serial                  int32    `json:"serial"`
        Checksum                string   `json:"checksum"`
+       BaseChecksum            string   `json:"base-checksum,omitempty"`
        Version                 string   `json:"version"`
        Timestamp               uint64   `json:"timestamp"`
        Booted                  bool     `json:"booted"`
```

Then use `BaseChecksum` if it's non-empty.

Reply:
Whoops, thanks, I thought I pushed that change up already.

Yeah, I was initially...not thinking about the checksum changing. I figured that out this morning after the `kernelType` e2e test added those kernel RPMs (and of course the checksum changed) and I was like "oh, yeah, duh". :smile: